### ISSUE TITLE
`clippy::single_match` does not report if a comment is on top of an arm

### ISSUE THREAD

**Issue by ojeda** [rust-lang/rust-clippy#14418]
### Summary

Starting with Rust 1.85.0, `clippy::single_match` apparently does not get emitted when there is a comment on top of an arm.

Is this intended?

### Lint Name

clippy::single_match

### Reproducer

I tried this code:

```rust
fn _f() {
    match Vec::<i32>::new().try_reserve(42) {
        Err(_) => (),
        // X
        Ok(()) => (),
    }
}
```

I expected to see this happen: Lint emitted as usual, e.g.

```text
warning: you seem to be trying to use `match` for destructuring a single pattern. Consider using `if let`
 --> <source>:2:5
  |
2 | /     match Vec::<i32>::new().try_reserve(42) {
3 | |         Err(_) => (),
4 | |         Ok(()) => (),
5 | |     }
  | |_____^ help: try: `if let Err(_) = Vec::<i32>::new().try_reserve(42) { () }`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_match
  = note: `#[warn(clippy::single_match)]` on by default

warning: 1 warning emitted
```

Instead, this happened: No lint emitted.

If one removes the `// X` line, or if one downgrades to 1.84.0, then the lint does get emitted.

Reproducer: https://godbolt.org/z/K7zoonG8j (Compiler Explorer supports Clippy now! :)

### Version

```text
rustc 1.85.0 (4d91de4e4 2025-02-17)
```

**Comment by samueltardieu** [comment]
This (non-lint) was introduced in #13765.

Maybe it would be a good idea to lint, but not mark the suggestion as `MaybeIncorrect` to prevent it from being auto-applied if it would drop comments.

**Comment by ojeda** [comment]
I see, thanks for the quick reply!

Yeah, if there are comments, then someone should manually inspect it and transform the comments as needed.
